doc/ImplementationGuide_SmartNIC.md

```diff
@@ -0,0 +1,35 @@
+# Implementation Guide SmartNIC
+
+## Introduction
+This is a guide to implement a generic OpenOffload solution using the gRPC code provided in this repository. 
+
+### Assumptions
+This guide there is a virtual network function (VNF) connected to a SmartNIC. The VNF is responsible for maintaining its own flow table of all sessions, and depending on internal polic, which is invisible to the SmartNIC instructing the SmartNIC to perform actions on the corresponding flows. The initial actions defined in the  gRPC module are:
```

Comment:
does it have to be a VNF or could it be a non VM based application or a CNF ? I have tended to use "Application" to avoid references to the virtualization/containerization approach.

Reply:
I have been thinking we need a glossary to keep terminology straight

Reply:
I think is needs to be a network function (CNF or VNF) as it is processing at the packet level and sending them back out again.  Not sure what type of generic socket based application would be used here. Would rather be more specific and open up later rather than the other way around.